src/plugins/clear_button/plugin.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Plugin: "clear_button" (selectize.js)
+ * Copyright (c) 2017 Alexey Kolotovchenkov & contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at:
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ *
+ * @author Alexey Kolotovchenkov <alkoloto@hotmail.com>
+ */
+
+Selectize.define('clear_button', function(options) {
+	var self = this;
+
+	options = $.extend({
+		label     : '&times;',
+		title     : 'Clear',
+		className : 'clear'
+	}, options);
+
+	self.setup = (function() {
+		var original = self.setup;
+		return function() {
+			original.apply(self, arguments);
+
+			var html = '<a href="javascript:void(0)" class="' + options.className + '" tabindex="-1" title="' + escape_html(options.title) + '">' + options.label + '</a>';
+
+			self.$control.append(html);
+
+			// add event listener
+			self.$control
+				.on('mousedown', 'a.' + options.className, function(e) {
+					// prevent toggling
+					e.preventDefault();
+					e.stopPropagation();
+				})
+				.on('click', 'a.' + options.className, function(e) {
+					e.preventDefault();
+					e.stopPropagation();
+					if (self.isLocked) return;
+
+					self.clear();
```

Comment:
*clear()* will not trigger the '*item_remove*' event (see pull request: https://github.com/selectize/selectize.js/pull/1311/commits/ea2e931fc1d1cb4eea530f951206f12401e80042#diff-a1e04fdc9fa0888b83d5b5401328081bR62)

Maybe it is on purpose, but just wanted to mention it, in case *deleteSelection()* is more appropriate.

Reply:
Actually `clear()` already triggers the event *clear*. And that might be enough.

Having the *item_remove* event being triggered automatically would be just convenience; probably one should listen `on("clear item_remove")` in case of wanting to execute the same callback for both events.
E.g. using the same callback for settings `onClear` and `onItemRemove` will achieve that.

Additionally, `deleteSelection` provides support for the `onDelete` callback, that is an actual useful feature (validate values before removal and eventually prevent the clear/remove operation).
I believe this should be still supported also when clearing or removing items (so that we can stick to what said in the previous paragraph).